src/DeFuncto.Core/Extensions/Tasks.cs

```diff
@@ -56,61 +107,134 @@ async Task<T> WrapInSemaphore(Func<Task<T>> f)
+    /// <param name="f">The asynchronous action to apply.</param>
+    /// <typeparam name="T">The type of the object.</typeparam>
+    /// <returns>Unit, asynchronously.</returns>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public static Task<Unit> RunAsync<T>(this T self, Func<T, Task> f) => self.Apply(async t =>
-    {
-        await f(t);
-        return Unit.Default;
-    });
-
-    [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public static Task<Unit> RunAsync<T>(this Task<T> self, Func<T, Task> f) => self.Map(async t =>
-    {
-        await f(t);
-        return Unit.Default;
-    });
-
+    public static Task<Unit> RunAsync<T>(this T self, Func<T, Task> f) =>
+        self.Apply(async t =>
+        {
+            await f(t);
+            return Unit.Default;
+        });
+
+    /// <summary>
+    /// Asynchronously applies an asynchronous action to the result of a task.
+    /// </summary>
+    /// <param name="self">The task to apply the asynchronous action to.</param>
+    /// <param name="f">The asynchronous action to apply.</param>
+    /// <typeparam name="T">The type of the task's result.</typeparam>
+    /// <returns>Unit, asynchronously.</returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public static Task<Unit> RunAsync<T>(this Task<T> self, Func<T, Task> f) =>
+        self.Map(async t =>
+        {
+            await f(t);
+            return Unit.Default;
+        });
+
+    /// <summary>
+    /// Asynchronously applies an asynchronous effectful function to the result of a task.
+    /// </summary>
+    /// <param name="self">The task to apply the asynchronous action to.</param>
+    /// <param name="f">The asynchronous effectful function to apply.</param>
+    /// <typeparam name="T">The type of the task's result.</typeparam>
+    /// <returns>Unit, asynchronously.</returns>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static Task<Unit> RunAsync<T>(this T self, Func<T, Task<Unit>> f) =>
         self.RunAsync(async t => { await f(t); });
 
-    [Pure]
+    /// <summary>
+    /// Performs a select in an IEnumerable returned by a task.
+    /// </summary>
+    /// <param name="self">The task returning the IEnumerable.</param>
+    /// <param name="func">The projection.</param>
+    /// <typeparam name="TIn">Type of the elements of the input.</typeparam>
+    /// <typeparam name="TOut">Type of the elements of the output.</typeparam>
+    /// <returns>The new enumerable, in a task.</returns>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public static Task<IEnumerable<TOut>> Select<TIn, TOut>(this Task<IEnumerable<TIn>> self, Func<TIn, TOut> func) =>
         self.Map(ienumerable => ienumerable.Select(func));
 
-    [Pure]
+    /// <summary>
+    /// Performs a select in an List returned by a task.
+    /// Might seem redundant since there's already an IEnumerable version, but the type inference in C#
```

Comment:
```suggestion
```
I would either move this observation to a remark or regullar comment or remove it.